src/bmi_cfe.c

```diff
@@ -2029,22 +2029,46 @@ static int Set_value_at_indices (Bmi *self, const char *name, int * inds, int le
      }   
         
     return BMI_SUCCESS;
+}*/
+
+// JG: 02.05.2024 - Implementing "standard" BMI definition 
+{
+    void * to = NULL;
+    int itemsize = 0;
+
+    if (self->get_value_ptr (self, name, &to) == BMI_FAILURE)
+        return BMI_FAILURE;
+
+    if (self->get_var_itemsize(self, name, &itemsize) == BMI_FAILURE)
+        return BMI_FAILURE;
+
+    { /* Copy the data */
+        size_t i;
+        size_t offset;
+        char * ptr;
+        for (i=0, ptr=(char*)src; i<len; i++, ptr+=itemsize) {
+            offset = inds[i] * itemsize;
+            memcpy ((char*)to + offset, ptr, itemsize);
+        }
+    }
+    return BMI_SUCCESS;
 }
 
 
 static int Set_value (Bmi *self, const char *name, void *array)
 {
-    // Avoid using set value, call instead set_value_at_index
+/*    // Avoid using set value, call instead set_value_at_index
     // Use nested call to "by index" version
 
     // Here, for now at least, we know all the variables are scalar, so
     int inds[] = {0};
 
     // Then we can just ...
-    return Set_value_at_indices(self, name, inds, 1, array);
+    return Set_value_at_indices(self, name, inds, 1, array);*/
```

Comment:
this only works for `doubles` and not for `array of doubles`, so it will potentially break the coupling with `soil_moisture_profile` which is used in rootzone-based AET.